freqtrade/strategy/strategy_updater.py

```diff
@@ -0,0 +1,213 @@
+import ast
+import os
+import shutil
+
+
+class strategy_updater:
+    name_mapping = {
+        'ticker_interval': 'timeframe',
+        'buy': 'enter_long',
+        'sell': 'exit_long',
+        'buy_tag': 'enter_tag',
+        'sell_reason': 'exit_reason',
+
+        'sell_signal': 'exit_signal',
+        'custom_sell': 'custom_exit',
+        'force_sell': 'force_exit',
+        'emergency_sell': 'emergency_exit',
+
+        # Strategy/config settings:
+        'use_sell_signal': 'use_exit_signal',
+        'sell_profit_only': 'exit_profit_only',
+        'sell_profit_offset': 'exit_profit_offset',
+        'ignore_roi_if_buy_signal': 'ignore_roi_if_entry_signal',
+        'forcebuy_enable': 'force_entry_enable',
+    }
+
+    function_mapping = {
+        'populate_buy_trend': 'populate_entry_trend',
+        'populate_sell_trend': 'populate_exit_trend',
+        'custom_sell': 'custom_exit',
+        'check_buy_timeout': 'check_entry_timeout',
+        'check_sell_timeout': 'check_exit_timeout',
+        # '': '',
+    }
+    # order_time_in_force, order_types, unfilledtimeout
+    otif_ot_unfilledtimeout = {
+        'buy': 'entry',
+        'sell': 'exit',
+    }
+
+    # create a dictionary that maps the old column names to the new ones
+    rename_dict = {'buy': 'enter_long', 'sell': 'exit_long', 'buy_tag': 'enter_tag'}
+
+    def start(self, strategy_obj: dict) -> None:
+        """
+        Run strategy updater
+        It updates a strategy to v3 with the help of the ast-module
+        :return: None
+        """
+
+        self.cwd = os.getcwd()
+        self.strategies_backup_folder = f'{os.getcwd()}user_data/strategies_orig_updater'
```

Comment:
please use [pathlib](https://docs.python.org/3/library/pathlib.html) for directory operations. You'll find that it has almost all methods available as os.<directoryoperation>.

This line will be wrong on windows (wrong slash) - so it'll either fail to open - or create a file in the cwd named as the string above (not sure - didn't use os.path for 2-3 years on windows).